tests/ignite/metrics/test_mutual_information.py

```diff
@@ -0,0 +1,147 @@
+import os
+from typing import Tuple
+
+import numpy as np
+import pytest
+import torch
+from scipy.special import softmax
+from scipy.stats import entropy
+from torch import Tensor
+
+import ignite.distributed as idist
+
+from ignite.engine import Engine
+from ignite.exceptions import NotComputableError
+from ignite.metrics import MutualInformation
+
+
+def np_mutual_information(np_y_pred: np.ndarray) -> float:
+    prob = softmax(np_y_pred, axis=1)
+    marginal_ent = entropy(np.mean(prob, axis=0))
+    conditional_ent = np.mean(entropy(prob, axis=1))
+    return max(0.0, marginal_ent - conditional_ent)
```

Comment:
@kzkadc can we use scikit-learn mutual info metric as a reference ? 
https://scikit-learn.org/stable/modules/clustering.html#mutual-information-based-scores